### PR TITLE
Introduce config for asdf version manager

### PR DESCRIPTION
**Pull request by bwilczek**
Let `asdf` manage `python` version for the project.

### CODE REVIEW

**Comment by id-ilych** [comment]
@bwilczek poetry manifest includes the python version, are you sure we need a separate tool and a config for that?

**Review by id-ilych — APPROVED** [review]
Copilot thinks you're right - poetry can't install python versions, and asdf wouldn't read `pyproject.toml` so it seems we indeed have to have the `.tool-versions`.

File: README.md [hunk 1]
```diff
@@ -81,6 +81,23 @@ Easy onboarding, reach out if you need support.
...
+### Managing python version with `asdf`
+
+Install `asdf` from https://asdf-vm.com/
+
+```
+# install plugin for python version management
+asdf plugin add python
+
+# install the right version
+asdf install python 3.12.0
```
Comment (id-ilych) [inline README.md:93]: Do we need to specify the version explicitly, considering we have `.tool-versions`?

File: .tool-versions [hunk 2]
```diff
@@ -0,0 +1,2 @@
+python 3.12.0
```
Comment (id-ilych) [inline .tool-versions:1]: How about bumping to the latest `3.12` release?

**Review by id-ilych — APPROVED** [review]

File: .tool-versions (hunk 2 above)
Comment (bwilczek) [reply .tool-versions]: done

File: README.md (hunk 1 above)
Comment (bwilczek) [reply README.md]: As I just tested we don't. Update the docs.

**Pull request merged**